src/decoder.rs

```diff
@@ -236,6 +257,31 @@ impl DOBDecoder {
             .as_bytes()
             .into())
     }
+
+    // search on-chain decoder cell, directly by its tx_hash and out_index
+    fn fetch_decoder_binary_directly(
+        &self,
+        tx_hash: H256,
+        out_index: usize,
+    ) -> DecodeResult<Vec<u8>> {
+        let decoder_deployment_tx = self
```

Comment:
It's better to use this rpc [get_live_cell](https://github.com/nervosnetwork/ckb/tree/develop/rpc#method-get_live_cell)